src/pyupgrade/plugins/rewrite_mock_import.rs

```diff
@@ -0,0 +1,150 @@
+use rustpython_ast::{AliasData, Located, Stmt, StmtKind};
+
+use crate::ast::types::Range;
+use crate::ast::whitespace::indentation;
+use crate::autofix::Fix;
+use crate::checkers::ast::Checker;
+use crate::checks::{Check, CheckKind};
+use crate::isort::helpers::trailing_comma;
+use crate::isort::types::TrailingComma;
+
+/// Replaces a given statement with a string
+fn update_import(checker: &mut Checker, stmt: &Stmt, new_stmt: String) {
+    let mut check = Check::new(CheckKind::RewriteMockImport, Range::from_located(stmt));
+    if checker.patch(check.kind.code()) {
+        check.amend(Fix::replacement(
+            new_stmt,
+            stmt.location,
+            stmt.end_location.unwrap(),
+        ));
+    }
+    checker.add_check(check);
+}
+
+/// Returns the amount of new lines in a given statement
+fn new_line_count(checker: &Checker, stmt: &Stmt) -> usize {
+    let contents = checker
+        .locator
+        .slice_source_code_range(&Range::from_located(stmt));
+    let mut count: usize = 0;
+    // I tried using tokens here, but the count didn't work
+    for character in contents.as_ref().chars() {
+        if character == '\n' {
+            count += 1;
+        }
+    }
+    count
+}
+
+/// Create the new string for the import
+fn create_new_statement(
+    needed_imports: Vec<String>,
+    beginning: &str,
+    indent: &str,
+    multi_line: bool,
+    magic_comma: bool,
+) -> String {
+    let mut new_stmt = String::new();
+    // If this is a mulit line import, we need to add the beginning
+    if multi_line {
+        new_stmt.push_str(" (")
+    }
+    for (i, import) in needed_imports.iter().enumerate() {
+        // We NEVER want a comma before the first item (since this would be import,
+        // example)
+        if i != 0 {
+            new_stmt.push(',');
+        }
+        // If this is a multi line import, we need to go to the next line and add an
+        // import
+        if multi_line {
+            new_stmt.push('\n');
+            new_stmt.push_str(indent);
+        }
+        // If this is multiline we will need an additonaly indent of 4 spaces beyond the
+        // `indent`
+        let gap = if multi_line { "    " } else { " " };
```

Comment:
For example, this assumes four-space indent, but LibCST would probably let you preserve whatever indent was already present (unless I'm misunderstanding this).

Reply:
This adds 4 spaces on top of any indent, so it still works. However I did not know about LibCST, and will definitely try it.

Reply:
Give LibCST a shot for this change and let me know how it goes. I'm here to answer any questions. There's definitely a learning curve (it uses a "concrete syntax tree" so the representation is a bit different and more detailed), but it's very powerful and could help make a lot of this stuff "safer" and easier to reason about.